src/components/Address/index.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import ENS from 'ethereum-ens'
+
+function useDebounce(value, timeout) {
+  const [debounceValue, setDebounceValue] = useState(value)
+
+  useEffect(
+    function() {
+      const timer = setTimeout(function() {
+        setDebounceValue(value)
+      }, timeout)
+
+      return function() {
+        clearTimeout(timer)
+      }
+    },
+    [value]
+  )
+  return debounceValue
+}
+
+async function resolveENSName(value, provider) {
+  const ethereumRegEx = /^0x[a-fA-F0-9]{40}$/
+  const ens = new ENS(provider)
+  if (ethereumRegEx.test(value)) {
+    return await ens
+      .reverse(value)
+      .name()
+      .then(async result => {
+        const address = await ens.resolver(result).addr()
```

Comment:
What happens if this errors?

Reply:
There is a catch block to take care of the errors already. In all the times I've tested it to break, it

Reply:
Yh think you might be right, if the promise errors at any point the final catch block should handle this